spec/features/authentication_spec.rb

```diff
@@ -1,28 +1,55 @@
 require 'rails_helper'
 
-describe "the signin process", :type => :feature do
-  let(:user) { User.create(:email => 'user@example.com', :password => 'password') }
+describe 'Authentication' do
+  let(:user) { create(:user) }
+  let(:admin) { create(:admin) }
 
-  scenario "Signs me in" do
-    visit root_path
-    click_link 'Login'
-    fill_in 'Email', with: user.email
-    fill_in 'Password', with: user.password
-    click_button 'Log in'
-    expect(current_path).to eq(root_path)
-    expect(page).to have_content('Signed in successfully')
+  context 'Guest user' do
+    scenario "Can sign up" do
+      visit new_user_registration_path
+      fill_in "Email", with: 'guest@example.com'
+      fill_in "Password", with: 'password'
+      fill_in "Password confirmation", with: 'password'
+      click_button "Sign up"
+
+      expect(page).to have_content('Welcome! You have signed up successfully.')
+    end
+
+    scenario "Can not access admin area" do
+      visit admin_root_path
+
+      expect(current_path).to eq(new_user_session_path)
+      expect(page).to have_content("Sorry, you just don't have access rights!")
+    end
   end
 
-  scenario "Signs me out" do
-    sign_in(user)
+  context 'Regular user' do
+    scenario "Can sign in and out" do
+      sign_in(user)
+      expect(page).to have_content('Signed in successfully')
 
-    click_link 'Sign out'
-    expect(page).not_to have_content('Sign out')
-    expect(page).to have_content('Signed out successfully')
+      click_link 'Sign out'
+      expect(page).to have_content('Signed out successfully')
+    end
+
+    scenario "Can not access admin area" do
+      sign_in(user)
+      visit admin_root_path
+      expect(page).to have_content("Sorry, you just don't have access rights!")
+    end
+  end
+
+  context 'Admin user' do
+    scenario "Can access admin area" do
+      sign_in(admin)
+      expect(current_path).to eq(admin_root_path)
+      expect(page).to have_content('Signed in successfully')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.